src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets

```diff
@@ -916,6 +916,8 @@ Copyright (c) .NET Foundation. All rights reserved.
         <NuGetAudit>$(NuGetAudit)</NuGetAudit>
         <NuGetAuditLevel>$(NuGetAuditLevel)</NuGetAuditLevel>
         <NuGetAuditMode>$(NuGetAuditMode)</NuGetAuditMode>
+        <SdkAnalysisLevel>$(SdkAnalysisLevel)</SdkAnalysisLevel>
```

Comment:
I don't fully understand why NuGet should have the SDKAnalysisLevel property ready for future consumption. I have a few questions, such as why we need these properties in nuget.exe and what future requirements we have that necessitate this property. Given that this is a new feature, I think we need a spec to fully understand the motivation behind this change. I was under the impression that we could use the new SDK analysis property similar to Andy's changes in the [commit](https://github.com/NuGet/NuGet.Client/commit/073a073d13b319f2d1d96af29d77f9dacd759053).

Reply:
Nigusu already linked the design spec in the PR description: https://github.com/dotnet/designs/pull/308

From NuGet's point of view, all new warnings (for example any future pack warnings, or any future restore warnings that aren't enabled only when a new feature is enabled), and any changes to defaults (like something that was previously a warning and we want to change to an error, or a feature that was previously opt-in and we want to change to opt-out) needs to be gated behind SdkAnalysisLevel

Reply:
Thanks for the response, Andy. I noticed the design spec in the PR description and have already read it.

If we want to introduce a breaking change, we should hardcode the SDKAnalysisLevel property in the code as needed so that customers can opt in or out. I felt that your approach in [this commit](https://github.com/NuGet/NuGet.Client/commit/073a073d13b319f2d1d96af29d77f9dacd759053) is much cleaner and easier to maintain than hardcoding the SDK version across the codebase as needed. We always have some escape hatches for features via environment variables, NuGet.Config (for example, allowing insecure connections), Visual Studio feature flags, or NoWarns. Now, we are adding another option. Hence, I'm double-checking and thought a spec would have helped clarify.

Reply:
The property value being propagated is just implementation detail. 

Not every behavior change is a project specific configuration change. 

Say for example, we change when NU1605 is raised, say it's raised in some additional scenarios. 
That logic will never be part of the nuget.targets, so it's not fair to set that restriction.